packages/react-components/react-table/src/components/DataGridHeaderCell/useDataGridHeaderCell.ts

```diff
@@ -19,13 +19,15 @@ export const useDataGridHeaderCell_unstable = (
   props: DataGridHeaderCellProps,
   ref: React.Ref<HTMLElement>,
 ): DataGridHeaderCellState => {
+  let { sortable = false } = props;
   const columnId = useColumnIdContext();
-  const { sortable } = useTableContext();
+  const { sortable: gridSortable } = useTableContext();
   const toggleColumnSort = useDataGridContext_unstable(ctx => ctx.sort.toggleColumnSort);
+  const columns = useDataGridContext_unstable(ctx => ctx.columns);
+  sortable = gridSortable || columns.find(c => c.columnId === columnId)?.compare.length !== 0;
```

Comment:
This would avoid returning a non-primitive from the DataGridContext (which avoids a performance regression)

```ts
  const columnSortable = useDataGridContext_unstable(ctx =>
    ctx.getSortableColumns().find(column => column.columnId === columnId),
  );
  const { sortable: gridSortable } = useTableContext();
  const toggleColumnSort = useDataGridContext_unstable(ctx => ctx.sort.toggleColumnSort);
  const { sortable = gridSortable ?? columnSortable } = props;
  const sortDirection = useDataGridContext_unstable(ctx =>
    sortable ? ctx.sort.getSortDirection(columnId) : undefined,
  );
  const r

```

Reply:
Playing aroud a bit with your PR this might be the best way to go

```ts
const isColumnSortable = (columns: TableColumnDefinition<any>[], columnId: TableColumnId) => {
  return !!columns.filter(c => c.compare.length !== 0 && c.columnId === columnId).length;
};

const useColumnSortable = (sortableProp: boolean | undefined) => {
  const columnId = useColumnIdContext();
  const columnSortable = useDataGridContext_unstable(ctx =>
    isColumnSortable(ctx.columns, columnId)
  );
  const { sortable: gridSortable } = useTableContext();

  return sortableProp ?? gridSortable ?? columnSortable;
};

// useDataGridHeaderCell_unstable
useColumnSortable(props.sortable)
```